packages/firebase_storage/firebase_storage/windows/firebase_storage_plugin.cpp

```diff
@@ -0,0 +1,807 @@
+      std::move(handler));
+
+  result(channelName);
+}
+
+void FirebaseStoragePlugin::ReferencePutString(
+    const PigeonStorageFirebaseApp& pigeon_app,
+    const PigeonStorageReference& pigeon_reference, const std::string& data,
+    int64_t format, const PigeonSettableMetadata& settable_meta_data,
+    int64_t handle, std::function<void(ErrorOr<std::string> reply)> result) {
+  Storage* cpp_storage =
+      GetCPPStorageFromPigeon(pigeon_app, pigeon_reference.bucket());
+  controllers_[handle] = std::make_unique<Controller>();
+
+  auto handler = std::make_unique<PutDataStreamHandler>(
+      cpp_storage, pigeon_reference.full_path(), &data, data.size(),
+      controllers_[handle].get());
+
+  std::string channelName = RegisterEventChannel(
+      kStorageMethodChannelName + "/" + kStorageTaskEventName,
+      std::move(handler));
+
+  result(channelName);
+}
+
+void FirebaseStoragePlugin::ReferencePutFile(
+    const PigeonStorageFirebaseApp& pigeon_app,
+    const PigeonStorageReference& pigeon_reference,
+    const std::string& file_path,
+    const PigeonSettableMetadata& settable_meta_data, int64_t handle,
+    std::function<void(ErrorOr<std::string> reply)> result) {
+  Storage* cpp_storage =
+      GetCPPStorageFromPigeon(pigeon_app, pigeon_reference.bucket());
+  controllers_[handle] = std::make_unique<Controller>();
+
+  auto handler = std::make_unique<PutFileStreamHandler>(
+      cpp_storage, pigeon_reference.full_path(), std::move(file_path),
+      controllers_[handle].get());
+
+  std::string channelName = RegisterEventChannel(
+      kStorageMethodChannelName + "/" + kStorageTaskEventName,
+      std::move(handler));
+
+  result(channelName);
+}
+
+void FirebaseStoragePlugin::ReferenceDownloadFile(
+    const PigeonStorageFirebaseApp& pigeon_app,
+    const PigeonStorageReference& pigeon_reference,
+    const std::string& file_path, int64_t handle,
+    std::function<void(ErrorOr<std::string> reply)> result) {
+  Storage* cpp_storage =
+      GetCPPStorageFromPigeon(pigeon_app, pigeon_reference.bucket());
+  controllers_[handle] = std::make_unique<Controller>();
+
+  auto handler = std::make_unique<GetFileStreamHandler>(
+      cpp_storage, pigeon_reference.full_path(), std::move(file_path),
+      controllers_[handle].get());
+
+  std::string channelName = RegisterEventChannel(
+      kStorageMethodChannelName + "/" + kStorageTaskEventName,
+      std::move(handler));
+
+  result(channelName);
+}
+
+void FirebaseStoragePlugin::ReferenceUpdateMetadata(
+    const PigeonStorageFirebaseApp& app,
+    const PigeonStorageReference& reference,
+    const PigeonSettableMetadata& metadata,
+    std::function<void(ErrorOr<PigeonFullMetaData> reply)> result) {
+  StorageReference cpp_reference =
+      GetCPPStorageReferenceFromPigeon(app, reference);
+  Metadata cpp_meta;
+  GetMetadataFromPigeon(metadata, &cpp_meta);
+  Future<Metadata> future_result = cpp_reference.UpdateMetadata(cpp_meta);
+  ::Sleep(1);
```

Comment:
Same thing for the Sleep here, adding a comment would be nice

Reply:
added the comments